builder/optimizations.cc

```diff
@@ -796,8 +797,24 @@ class in_place_aliaser : public stmt_mutator {
     }
   }
 
+  bool fold_factors_strides_same(const std::vector<dim_expr>& alloc_dims, const std::vector<dim_expr>& alias_dims) {
+    if (alloc_dims.size() != alias_dims.size()) {
+      return !(std::any_of(alloc_dims.begin(), alloc_dims.end(),
+          [&](const dim_expr& i) { return i.stride.defined() || i.fold_factor.defined(); }));
+    }
+    for (std::size_t ix = 0; ix < alloc_dims.size(); ++ix) {
+      if ((alloc_dims[ix].stride.defined() || alias_dims[ix].stride.defined()) &&
```

Comment:
You can ignore the case where `alias_dims[d].stride` is defined but `alloc_dims[d].stride` is not, i.e. you only need to check if the strides are equal if `alloc_dims[d].stride` is defined.

The idea here is that `alloc_dims` are getting replaced by a different allocation. If `alloc_dims` has an explicit stride, it must not change. But if it doesn't, it doesn't matter, because whatever used it should be fine with whatever the alias has (whether it is explicit or not).

Obviously the same applies to the fold_factor too.